File: oto-shop/src/main/java/com/otostore/oto_shop/Service/CatalogService.java
package com.otostore.oto_shop.Service;

import com.otostore.oto_shop.Entity.Catalog;

import java.util.List;
import java.util.Optional;

public interface CatalogService {

    // Lấy tất cả danh mục
    List<Catalog> getAllCatalogs();

    // Lấy danh mục theo ID
    Optional<Catalog> getCatalogById(Integer id);

    // Thêm danh mục mới
    Catalog createCatalog(Catalog catalog);

    // Cập nhật danh mục
    Catalog updateCatalog(Integer id, Catalog catalog);

    // Xóa danh mục
    void deleteCatalog(Integer id);
}

File: frontend/src/services/catalogService.js
import api from './api';

const catalogService = {
    // Lấy tất cả danh mục
    getAllCatalogs: () => {
        return api.get('/catalogs');
    },

    // Lấy danh mục theo ID
    getCatalogById: (id) => {
        return api.get(`/catalogs/${id}`);
    },
};

export default catalogService;